### PR TITLE
Update RawSerial and Serial class references/descriptions

### PR DESCRIPTION
**Pull request by yennster**

### CODE REVIEW

File: docs/api/drivers/Serial.md [hunk 1]
```diff
@@ -12,14 +12,15 @@ One of the serial connections uses the Arm Mbed USB port, allowing you to easily
...
 
-Serial channels have a number of configurable parameters:
+Serial channels have the following configurable parameters:
 
-  - _Baud Rate_ - There are a number of standard baud rates ranging from a few hundred bits per second, to megabits per second. The default setting for a serial connection on the Mbed microcontroller is 9600 baud.
-  - _Data length_ - Data transferred can be either 7 or 8 bits long. The default setting for a serial connection on the Mbed microcontroller is 8 bits.
-  - _Parity_ - You can add an optional parity bit. The parity bit will be automatically set to make the number of 1s in the data either odd or even. Parity settings are *Odd*, *Even* or *None*. The default setting for a serial connection on the Arm Mbed microcontroller is None.
-  - _Stop Bits_ - After data and parity bits have been transmitted, one or two stop bits are inserted to "frame" the data. The default setting for a serial connection on the Mbed microcontroller is one stop bit.
+  - _Tx and Rx Pin_ - The physical serial transmit and receive pins. You can specify a Tx or Tx pin as Not Connected (NC) to get Simplex communication, or specify both to get full duplex.
```
Comment (AnotherButler) [inline docs/api/drivers/Serial.md]: Query: Should this be "Tx or Rx" instead of "TX or TX"?

File: docs/api/drivers/Serial.md [hunk 2]
```diff
@@ -12,14 +12,15 @@ One of the serial connections uses the Arm Mbed USB port, allowing you to easily
...
+  - _Baud Rate_ - Standard baud rates range from a few hundred bits per second to megabits per second. The default setting for a serial connection on the Mbed microcontroller is 9600 baud.
+  - _Data length_ - Transferred data can be either 7 or 8 bits long. The default setting for a serial connection on the Mbed microcontroller is 8 bits.
```
Comment (AnotherButler) [inline docs/api/drivers/Serial.md]: Query: Do we need to specify Mbed microcontroller here? Can we just say "the microcontroller"?

File: docs/api/drivers/Serial.md [hunk 3]
```diff
@@ -12,14 +12,15 @@ One of the serial connections uses the Arm Mbed USB port, allowing you to easily
...
+  - _Parity_ - You can add an optional parity bit. The Serial object automatically sets the parity bit to make the number of 1s in the data either odd or even. Parity settings are *Odd*, *Even* or *None*. The default setting for a serial connection on the Arm Mbed microcontroller is None.
+  - _Stop Bits_ - After the transmission of data and parity bits, the Serial object inserts one or two stop bits to "frame" the data. The default setting for a serial connection on the Mbed microcontroller is one stop bit.
```
Comment (AnotherButler) [inline docs/api/drivers/Serial.md]: Query: Do we need to specify Arm Mbed microcontroller here? Can we just say "the microcontroller"?

**Comment by yennster** [comment]
@AnotherButler done 👍

**Pull request merged**